src/components/GuardedRoute.js

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { Redirect } from 'react-router-dom';
+import { checkToken } from '../lib/TokenService';
+
+const GuardedRoute = (Component) => ({ match }) => {
+  if (!checkToken()) {
+    return <Redirect to="/" />;
+  } else {
+    return <Component match={match} />;
+  }
+};
```

Comment:
I didn't know about route guarding, which seems great for protecting unrestricted access ! (now I know, thanks! 👍  ) 
Just one tiny comment, I guess I'd use a ternary operator here but the function is too short anyway so it won't change much 😄

Reply:
Yeah def a ternary operator will fit much better here 👌

Reply:
Thanks for the suggestion Deborah. it's been refactored based on your suggestion 👌

Reply:
Very interesting and cool approach. I love it. 👏